src/Command/Binary/ReadBuffer.php

```diff
@@ -161,12 +161,16 @@ protected function handleBody($buffer)
     {
         // split buffer into arguments
         $args = array_keys($this->currentCommand->getAll());
-        $argv = strlen($buffer) ? explode(CommandInterface::ARGUMENT_DELIMITER, $buffer, count($args)) : [];
+        $argv = strlen($buffer) ? explode(CommandInterface::ARGUMENT_DELIMITER, $buffer) : [];
 
         // validate argument-count vs expected argument count
-        if (count($args) != count($argv)) {
+        if (count($args) > count($argv)) {
             throw new ProtocolException("Invalid package-header: header.size bytes did not contain full package body");
         }
+        // ignore any data after the last allowed \0. See https://github.com/gearman/gearmand/issues/108
```

Comment:
We can only ignore `\0` after the last argument, if it's not data. Binary data might be padded with 0-n \0.